flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpillingAdaptiveSpanningRecordDeserializer.java

```diff
@@ -634,8 +638,12 @@ private FileChannel createSpillingChannel() throws IOException {
 			for (int attempt = 0; attempt < maxAttempts; attempt++) {
 				String directory = tempDirs[rnd.nextInt(tempDirs.length)];
 				spillFile = new File(directory, randomString(rnd) + ".inputchannel");
-				if (spillFile.createNewFile()) {
-					return new RandomAccessFile(spillFile, "rw").getChannel();
+				try {
+					if (spillFile.createNewFile()) {
+						return new RandomAccessFile(spillFile, "rw").getChannel();
+					}
+				} catch (IOException e) {
+					LOG.warn("Exception while find a unique file name for the spilling channel, try it again.", e);
```

Comment:
You should give more context and improve the message a bit, maybe something like
```
LOG.warn("Failed to create spilling channel on {}, trying again.", spillFile.getCanonicalPath(), e);
```